app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  CONFIRMATION_TOKEN_EXPIRES_IN = 5.minutes
+
+  enum role: { admin: 0, doctor: 1, patient: 2 }
+  enum gender: { not_specified: 0, female: 1, male: 2 }
+  enum status: { pending: 0, active: 1, deleted: 2 }
+
+  after_create :generate_confirmation_token, if: -> { pending? }
+  after_create :send_confirmation_email, if: -> { pending? && confirmation_token.present? }
+
+  attr_accessor :confirmation_token
+
+  has_secure_password
+
+  validates :first_name, presence: true, length: { maximum: 128 }
+  validates :last_name, presence: true, length: { maximum: 128 }
+  validates :gender, :date_of_birth, :role, :password_digest, presence: true
+  validates :email, presence: true, uniqueness: true, format: { with: URI::MailTo::EMAIL_REGEXP, allow_blank: true }
```

Comment:
`presence: true` and `allow_blank: tru`e isn't this contradictory?

Reply:
@kth-wtag apu, as far as I know, `allow_blank` skips the format checking using the Regex if the field is blank, but, presence true is still checked even if the format checking is skipped. Also, `allow_blank` is necessary for cases where the field is blank or empty to avoid getting multiple errors (both not present and invalid format). Thus, I think these two are not contradictory to each other.

Reply:
> @kth-wtag apu, as far as I know, `allow_blank` skips the format checking using the Regex if the field is blank, but, presence true is still checked even if the format checking is skipped. Also, `allow_blank` is necessary for cases where the field is blank or empty to avoid getting multiple errors (both not present and invalid format). Thus, I think these two are not contradictory to each other.

when giving `presence: true` the field can never be blank or empty. If it is a empty string then only `format` is enough to check the format of email.